python/ray/types.py

```diff
@@ -7,6 +7,25 @@
 
 # TODO(ekl) this is a dummy generic ref type for documentation purposes only.
 # We should try to make the Cython ray.ObjectRef properly generic.
+# NOTE(sang): Looks like using Generic in Cython is not currently possible.
+# We should update Cython > 3.0 for this.
 @PublicAPI
 class ObjectRef(Generic[T]):
     pass
+
+
+@PublicAPI(stability="beta")
+class StreamingObjectRefGenerator(Generic[T]):
```

Comment:
Note: Currently, Cython cannot define generic inherited classes. WE need to upgrade it to support it